src/components/StateProvider/StateProvider.js

```diff
@@ -0,0 +1,20 @@
+import React, { createContext, useContext, useReducer } from 'react';
+import {
+  func, objectOf, object, instanceOf,
+} from 'prop-types';
+
+export const StateContext = createContext();
+
+export const StateProvider = ({ reducer, initialState, children }) => (
+  <StateContext.Provider value={useReducer(reducer, initialState)}>
+    {children}
+  </StateContext.Provider>
+);
+
```

Comment:
add a ref for this part, better for us to review

Reply:
add a ref?

Reply:
> add a ref for this part, better for us to review

add a ref?

Reply:
external reference source for reviewer

Reply:
sure, I will add the medium article that I referred to the PR.